library/src/androidTest/java/org/xmtp/android/library/ConversationsTest.kt

```diff
@@ -253,4 +262,101 @@ class ConversationsTest {
             assertTrue(topics.contains(convo.topic))
         }
     }
+
+    @Test
+    fun testStreamsAndMessages() = runBlocking {
+        val messages = mutableListOf<String>()
+        val alixGroup = alixClient.conversations.newGroup(listOf(caroClient.address, boClient.address))
+        val caroGroup2 = caroClient.conversations.newGroup(listOf(alixClient.address, boClient.address))
+
+
+        // Ensure all clients sync before sending/receiving
+        alixClient.conversations.syncAllConversations()
+        caroClient.conversations.syncAllConversations()
+        boClient.conversations.syncAllConversations()
+
+        val boGroup = boClient.findGroup(alixGroup.id)!!
+        val caroGroup = caroClient.findGroup(alixGroup.id)!!
+        val boGroup2 = boClient.findGroup(caroGroup2.id)!!
+        val alixGroup2 = alixClient.findGroup(caroGroup2.id)!!
+
+        val caroJob = launch(Dispatchers.IO) {
+            println("Caro is listening...")
+            try {
+                withTimeout(60.seconds) { // Ensure test doesn't hang indefinitely
+                    caroClient.conversations.streamAllMessages()
+                        .take(80) // Stop after receiving 80 messages
+                        .collect { message ->
+                            synchronized(messages) { messages.add(message.body) }
+                            println("Caro received: ${message.body}")
+                        }
+                }
+            } catch (e: TimeoutCancellationException) {
+                println("Timeout reached for caroJob")
+            }
+        }
+
+        delay(1000)
+
+        // Simulate message sending in multiple threads
+        val alixJob = launch(Dispatchers.IO) {
+            println("Alix is sending messages...")
+            repeat(20) {
+                val message = "Alix Message $it"
+                alixGroup.send(message)
+                alixGroup2.send(message)
+                println("Alix sent: $message")
+            }
+        }
+
+        val boMessageJob = launch(Dispatchers.IO) {
+            println("Bo is sending messages..")
+            repeat(10) {
+                val message = "Bo Message $it"
+                boGroup.send(message)
+                boGroup2.send(message)
+                println("Bo sent: $message")
+            }
+        }
+
+//        val boSpamJob = launch(Dispatchers.IO) {
+//            println("Bo is sending spam groups..")
+//            repeat(10) {
+//                val spamMessage = "Bo Message $it"
+//                val group = boClient.conversations.newGroup(listOf(caroClient.address))
+//                group.send(spamMessage)
+//                println("Bo spam: $spamMessage")
+//            }
+//        }
```

Comment:
@insipx I added this test and it passes about 80% of the time. Occasionally misses the first sent message but otherwise is pretty reliable. 
However when I add in this group creation it doesn't seem to stream messages from new groups? If I get added to a group I shouldn't have to resubscribe to `streamAllMessages` right? Or is this expected behavior.

Reply:
you should not have to re-subscribe, the stream should pick up on the new group and add it. in `xmtp_mls` we have Eve create a group and send a message to test this out, if its not working then there's a bug somewhere. This test seems more involved then the libxmtp one, so ill try adding in more things to the test in xmtp_mls and see if i can reproduce that behavior